apps/aecore/src/aec_conductor.erl

```diff
@@ -217,6 +220,7 @@ handle_call(reinit_chain, _From, State1) ->
                 State3 = kill_all_workers(State2),
                 epoch_mining:info("Mining started"),
                 start_mining(State3#state{mining_state = running,
+                                          new_candidate_available = false,
```

Comment:
You need to stop the candidate generation here as well. And prevent the race.

Reply:
Or rather, kill the candidate geration to get a fresh start, and prevent the race.